### PR TITLE
Goonchat

### PR DESCRIPTION
**Pull request by tigercat2000**
https://github.com/d3athrow/vgstation13/pull/9043/files

:cl:
rscadd: Ported goon HTML chat from /vg/ / Goon
tweak: All uses of \black have been removed
tweak: All uses of \icon have been replaced with bicon()
/:cl:

### CODE REVIEW

**Comment by Spacemanspark** [comment]
Does saving logs still work the same way?

**Comment by tigercat2000** [comment]
@Spacemanspark no

**Comment by Spacemanspark** [comment]
Can you explain how it works, then?
Couldn't figure it out on Goonstation awhile back.

**Comment by tigercat2000** [comment]
you go to the little gear icon and hit `Save chat log`
then copy-paste from the window that appears a few seconds later

**Comment by Spacemanspark** [comment]
Oh, didn't see it before on the menu there, my bad.

File: code/modules/mob/living/carbon/human/examine.dm [hunk 1]
```diff
@@ -41,7 +41,7 @@
 		t_is = "are"
 	else
 		if(icon)
-			msg += "\icon[icon] " //fucking BYOND: this should stop dreamseeker crashing if we -somehow- examine somebody before their icon is generated
+			msg += "[bicon(icon)] " //fucking BYOND: this should stop dreamseeker crashing if we -somehow- examine somebody before their icon is generated
```
Comment (Tastyfish) [inline code/modules/mob/living/carbon/human/examine.dm]: ```msg += "[bicon(icon(icon, dir=SOUTH))] "```
should fix the dir frames problem you had

**Comment by Fox-McCloud** [comment]
Does this include the "jump to message" feature where you click it and it pushes you all the way to the bottom if you're in the middle of the box?

**Comment by tigercat2000** [comment]
@Fox-McCloud i think i saw something like that

**Comment by Spacemanspark** [comment]
The server exploded Tiger.
We blame you

**Comment by Spacemanspark** [comment]
It couldn't handle the Goon

**Pull request merged**